packages/dev/core/src/Cameras/arcRotateCamera.ts

```diff
@@ -1078,21 +1079,22 @@ export class ArcRotateCamera extends TargetCamera {
      * Defines the target the camera should look at.
      * This will automatically adapt alpha beta and radius to fit within the new target.
      * Please note that setting a target as a mesh will disable panning.
-     * @param target Defines the new target as a Vector or a mesh
+     * @param target Defines the new target as a Vector or a transform node
      * @param toBoundingCenter In case of a mesh target, defines whether to target the mesh position or its bounding information center
      * @param allowSamePosition If false, prevents reapplying the new computed position if it is identical to the current one (optim)
      * @param cloneAlphaBetaRadius If true, replicate the current setup (alpha, beta, radius) on the new target
      */
-    public setTarget(target: AbstractMesh | Vector3, toBoundingCenter = false, allowSamePosition = false, cloneAlphaBetaRadius = false): void {
+    public setTarget(target: TransformNode | Vector3, toBoundingCenter = false, allowSamePosition = false, cloneAlphaBetaRadius = false): void {
         cloneAlphaBetaRadius = this.overrideCloneAlphaBetaRadius ?? cloneAlphaBetaRadius;
 
-        if ((<any>target).getBoundingInfo) {
+        const canTakeHostPath = (!toBoundingCenter || (toBoundingCenter && (<any>target).getBoundingInfo)) && (target as TransformNode).computeWorldMatrix;

```

Comment:
if toBoundingCenter is true and target is a transformNode, I guess we will fall into the else case where target is being used as a Vector3 which may result in a crash ?